fg21sim/extragalactic/pointsources/flux.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2016 Zhixian MA <zxma_sjtu@qq.com>
+# MIT license
+
+"""
+A class to calculate the brightness temperature of point sources is
+defined, based on the works of Wang et al. and Willman et al.
+
+[1] Wang J et al.,
+    "How to Identify and Separate Bright Galaxy Clusters from the
+    Low-frequency Radio Sky?",
+    2010, ApJ, 723, 620-633.
+    http://adsabs.harvard.edu/abs/2010ApJ...723..620W
+[2] Wilman et al.,
+    "A semi-empirical simulation of the extragalactic radio continuum
+    sky for next generation radio telescopes",
+    2008, MNRAS, 388, 1335-1348.
+    http://adsabs.harvard.edu/abs/2008MNRAS.389.1335W
+"""
+
+import numpy as np
+
+import astropy.constants as ac
+import astropy.units as au
+
+
+class Flux:
+    """
+    To calculate the flux and surface brightness of the point sources
+    accordingly to its type and frequency
+
+    Parameters
+    ----------
+    freq: float
+        The frequency
+    ClassType: int
+        The type of point source, which is default as 1.
+        | ClassType | Code |
+        |:---------:|:----:|
+        |    SF     |  1   |
+        |    SB     |  2   |
+        |  RQ AGN   |  3   |
+        |   FRI     |  4   |
+        |   FRII    |  5   |
+
+
+    Functions
+    ---------
+    genspec:
+        Generate the spectrum of the source at frequency freq.
+    calc_Tb:
+        Calculate the average surface brightness, the area of the source
+        should be inputed.
+    """
+
+    def __init__(self, freq=150, ps_type=1):
+        self.freq = freq
+        self.ps_type = ps_type
+
+    def gen_spec(self):
+        """Generate the spectrum """
```

Comment:
Split ``gen_spec()`` functionality into each PS classes, e.g., ``FRI.gen_spectrum()`` and ``SF.gen_spectrum()``.

Thus also avoid the annoying ``if``s on ``ps_type`` checks.